Chat: Add closure notice for Thanksgiving

jordwest opened the pull request:
See: p2UL9c-33A-p2

Adds a notice at the top of /help/contact informing chat-eligible users of the upcoming closure for the US Thanksgiving holiday. When the closure begins, the notice is automatically adjusted. When the closure ends, the notice disappears.

### How to test

1. Log in as a plan user who is eligible for live chat support.
1. Head to http://calypso.localhost:3000/help/contact, confirm that the **before closure** notice is displayed.
1. Change your system time to somewhere between the closure period, then reload the page. Confirm that the **during closure** notice is displayed.
1. Change your system time to some time after the closure period, then reload the page. Confirm that no notice is displayed.
1. Reset system time to normal.
1. Log in as a free user who is not eligible for chat. You should not see the notice.

### Before closure:

<img width="645" alt="screen shot 2016-11-18 at 4 14 39 pm" src="https://cloud.githubusercontent.com/assets/416133/20420631/c5bbd6e4-adaa-11e6-922b-504c44dc6d7e.png">

### During closure:

<img width="647" alt="screen shot 2016-11-18 at 4 15 21 pm" src="https://cloud.githubusercontent.com/assets/416133/20420638/cdaf37ec-adaa-11e6-9d6e-c269b859cfd8.png">

### After closure

<img width="652" alt="screen shot 2016-11-18 at 4 22 17 pm" src="https://cloud.githubusercontent.com/assets/416133/20420701/319463cc-adab-11e6-94e3-9e4c4180c199.png">

### Todo

- [x] Upcoming closure notice
- [x] Only show for chat-eligible users
- [x] Show different notice after closure has begun
- [x] Hide notice when we reopen

jordwest commented:
cc @andreabadgley @dllh 

for code review @omarjackman @southp

southp commented on client/me/help/help-contact-closure-notice/index.jsx:
```diff
@@ -0,0 +1,99 @@
+export default class HelpContactClosureNotice extends Component {
+	constructor() {
+		super();
+		this.state = { olark: olarkStore.get() };
```
As the up-stream component `HelpContact` is having `olarkStore` embedded and doing all the updating and listening acts, I am wondering if it was possible to pass the props you need from there and turn this component a dummy component.

However, it could make the data passing chain a bit long and dirty( HelpContact => HelpContactForm => HelpContactClosureNotice ). Thus, this is more of a question than a improvement comment. Just wanted to raise the thought :)

jordwest replied:
I had thought about that, but as you mentioned that would introduce data into an extra intermediary component (HelpContactForm) which doesn't otherwise need that data. Ideally we'd have this data in the redux store eventually anyway, in which case I'd just use a `connect` on HelpContactClosureNotice.

southp commented:
Tested the following:

* Logged-in as a eligible user:
    * Set the date before the closure and see the pre-notice: *CHECK!*
    * Set the date during the closure and see the during-notice: *CHECK!*
    * Set the date after the closure and don't see any closure notice: *CHECK!*
* Logged-in as a non-eligible user and don't see any closure notices no matter which time ranges I set: *CHECK!*

Though I've left a question, the code in general looks and works good to me. However I don't think I am the right person to give an authoritative comment for the text and the design.

cc @andreabadgley and @evilluendas for them :)

andreabadgley commented:
The wording looks good to me -- thanks @jordwest

omarjackman commented:
The code here looks good to me however the name of the components suggest something thats more generic than a Thanksgiving notice. Maybe if the the notice component was generic and the Thanksgiving text was passed in to it somehow it'd help. I'm just thinking about future notices for Christmas and New Years that might come up.

jordwest commented:
Thanks for the reviews! I'll go ahead and merge this since it's time critical, but I'll open another PR shortly to address the feedback.

jordwest commented:
@omarjackman: I've merged this, and opened #9533 which I think addresses your concerns. Passing in the message text was a bit complicated, since the message changes based on the current date and the logic for checking that is inside the component. Instead I opted to use a 'reason' prop and a switch statement, then defined the messages inside the component.
